### PR TITLE
[FIX] account: chart template upgrade

### PR DESCRIPTION
**Pull request by agrabsi**
**Description of the issue/feature this PR addresses:**

When upgrading the chart template we pass an update variable because `_load_records` is supposed to take an update parameter to indicate if we are updating or installing for the first time. if this flag is missing we will ignore `noupdate=True` raising the problems described below.

To fix this issue we need to pass a `update=True` argument to the `try_loading` call in the upgrade script
https://github.com/odoo/upgrade/blob/master/migrations/l10n_mx/saas~16.2.2.0/post-migrate.py#L19
another PR in odoo/upgrade will be made for that once the solution in the current one is approved.

**Current behavior before PR:**
- existing `account.account` record with no xmlid pointing to them are ignored, the orm will try to create the record again triggering a duplicate error on key `(code, company_id)`
- Ignore noupdate true: if a customer modified a record their changes will be overwritten, which also can lead to duplicate errors

**Desired behavior after PR is merged:**
- Rename old records that will create a duplicate to "[code].old" so it doesn't clash anymore.
- create xmlids for existing account records instead of duplicating data

Let me know if something is wrong and shouldn't be done like that. I also need some help for another bug in here:
https://github.com/odoo/odoo/blob/saas-16.2/addons/stock_account/__init__.py#L29

`_load_data` should also pass `update=True` as the method definition suggests
```py
def _load_data(self, data, update=False):
```
otherwise we also have a problem with duplicates
```
psycopg2.errors.UniqueViolation: duplicate key value violates unique constraint "account_journal_code_company_uniq"
DETAIL:  Key (company_id, code)=(1, STJ) already exists.
```

I didn't find a way to know in that post-init hook if we are updating or installing so I'm not sure how to pass that boolean correctly at the right time.

### CODE REVIEW

**Comment by william-andre** [comment]
We are not using the noupdate flag because we actually want to update some fields but not others (for when legal changes are done in the tax report for instance)
Everything should be managed in `_pre_reload_data`

(also, tests are failing)

**Comment by agrabsi** [comment]
@william-andre I have updated the PR following your advice. Known problems related to `account.account` should now be correctly handled. I tested it on a couple of failing upgrades and it seems to work properly.

What do you think of the solution of adding a `.old` to the previously exiting records causing problems?

(tests are now passing)

File: addons/account/models/chart_template.py [hunk 1]
```diff
@@ -290,6 +290,29 @@ def tax_template_changed(tax, template):
...
+                    account = self.ref(xmlid, raise_if_not_found=False)
+                    if account:
+                        # If update causes code duplicate, adjust old.
+                        if account.code != values['code']:
+                            existing_account = self.env['account.account'].search([
+                                ('code', '=', values['code']),
+                                ('company_id', '=', company.id),
+                            ])
+                            if existing_account:
+                                existing_account.code = f"{existing_account.code}.old"
```
Comment (william-andre) [inline addons/account/models/chart_template.py:303]: Update the xml_id of `existing_account` to `xmlid` instead

**Comment by william-andre** [comment]
It would also be nice to add a test in `addons/account/tests/test_chart_template.py`

**Comment by agrabsi** [comment]
@william-andre done ! the tests should cover both previously failing cases

File: addons/account/tests/test_chart_template.py [hunk 2]
```diff
@@ -209,3 +209,32 @@ def local_get_data(self, template_code):
...
+                ('code', '=', '222221')
+            ])
+        standard_account.code = '111111'
+
+        # create a new account with the same code as an existing one
+        problematic_account = self.env['account.account'].create({
+                'code': '222221',
+                'name': 'problematic_account',
+                'company_id': self.company_1.id,
+            })
```
Comment (william-andre) [inline addons/account/tests/test_chart_template.py:226]:
```suggestion
        problematic_account = self.env['account.account'].create({
            'code': '222221',
            'name': 'problematic_account',
            'company_id': self.company_1.id,
        })
```

**Review by william-andre — COMMENTED** [review]
Looking good!
Just some nitpicking :)

**Review by william-andre — APPROVED** [review]
@robodoo r+

**Pull request closed without merging**